src/_pytest/python.py

```diff
@@ -493,13 +498,8 @@ def _genfunctions(self, name: str, funcobj) -> Iterator["Function"]:
         if not metafunc._calls:
             yield Function.from_parent(self, name=name, fixtureinfo=fixtureinfo)
         else:
-            # Add funcargs() as fixturedefs to fixtureinfo.arg2fixturedefs.
-            fm = self.session._fixturemanager
-            fixtures.add_funcarg_pseudo_fixture_def(self, metafunc, fm)
-
-            # Add_funcarg_pseudo_fixture_def may have shadowed some fixtures
-            # with direct parametrization, so make sure we update what the
-            # function really needs.
+            # Dynamic direct parametrization may have shadowed some fixtures,
```

Comment:
It means the `metafunc.parametrize()` calls that user does in module-specific or class-specific `pytest_generate_tests` hooks, as opposed to the parametrizations using `@pytest.mark.parametrize` decorator.

Reply:
Are you sure there's a difference between `@pytest.mark.parametrize` and module/class level `pytest_generate_tests` hooks? From a cursory look they are at the same level, they're all called together on the same `Metafunc` at the same time:

https://github.com/pytest-dev/pytest/blob/485c555812286003d9234b5818c9852c3256bccb/src/_pytest/python.py#L476-L491

Reply:
as extra caution, there is cases where users use module/function specific lists in parametrize calls,

Reply:
The differentiation between dynamic and non-dynamic parametrization and pruning dependency tree only when the former has taken place is done in #11243 , but the comment is correct in the sense that only dynamic ones may shadow some fixtures in the fixtureclosure as parametrization args introduced in the non-dynamic ones are already considered in computing fixture closure (using `ignore_args` arg to `getfixtureclosure`)

Reply:
> but the comment is correct in the sense that only dynamic ones may shadow some fixtures in the fixtureclosure as parametrization args introduced in the non-dynamic ones are already considered in computing fixture closure (using ignore_args arg to getfixtureclosure)

Ah, got it, thanks.

Let's avoid the word "dynamic", it can mean several things so it's not very clear. Instead something like your comments here would be very helpful.